syntax = "proto3";

package container_service;
option go_package = "github.com/cicadatesting/container-service/api";

import "google/protobuf/empty.proto";
import "google/protobuf/wrappers.proto";

service ContainerService {
    rpc StartContainer (StartContainerRequest) returns (google.protobuf.Empty);
    rpc StopContainer (StopContainerRequest) returns (google.protobuf.Empty);
}

message StartContainerRequest {
    string image = 1;
    string name = 2;
    repeated string command = 3;
    repeated string labels = 4;
    map<string, string> env = 5;
    repeated DockerVolume volumes = 6;
    google.protobuf.Int32Value hostPort = 7;
    google.protobuf.Int32Value containerPort = 8;
    google.protobuf.StringValue network = 9;
    google.protobuf.BoolValue createNetwork = 10;
}

message DockerVolume {
    string source = 1;
    string destination = 2;
}

message StopContainerRequest {
    string containerID = 1;
}
